server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -6139,4 +6140,78 @@ public void testMaxDocsOnReplica() throws Exception {
             IndexWriterMaxDocsChanger.restoreMaxDocs();
         }
     }
+
+    public void testCurrentVersionIsCommitted() throws IOException {
+        final AtomicLong globalCheckpoint = new AtomicLong(SequenceNumbers.NO_OPS_PERFORMED);
+        try (Store store = createStore()) {
+            EngineConfig config =
+                config(defaultSettings, store, createTempDir(), newMergePolicy(), null, null, globalCheckpoint::get);
+
+            store.createEmpty();
+            final String translogUUID = Translog.createEmptyTranslog(
+                config.getTranslogConfig().getTranslogPath(),
+                SequenceNumbers.NO_OPS_PERFORMED,
+                shardId,
+                primaryTerm.get()
+            );
+            store.associateIndexWithNewTranslog(translogUUID);
+
+            try (InternalEngine engine = createEngine(config)) {
+                engine.flush();
```

Comment:
I am not sure this will flush since there are no changes to flush? I think you need to call `flush(true, true);`